src/conn/conn_dhandle.c

```diff
@@ -17,7 +17,11 @@ __conn_dhandle_destroy(WT_SESSION_IMPL *session, WT_DATA_HANDLE *dhandle)
 {
 	WT_DECL_RET;
 
-	WT_WITH_DHANDLE(session, dhandle, ret = __wt_btree_discard(session));
+	if (WT_PREFIX_MATCH(dhandle->name, "file:"))
```

Comment:
We could do that for built-in data handle types, but data sources should (eventually) be able to use the same handle caching.  I guess we could have a generic type to cover external data sources?